hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/timeline/TimestampUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.timeline;
+
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.timeline.HoodieTimeline;
+import org.apache.hudi.common.util.ValidationUtils;
+
+public class TimestampUtils {
+
+  public static void validateForLatestTimestamp(HoodieTableMetaClient metaClient, String instantTime) {
+    // validate that the instant for which requested is about to be created is the latest in the timeline.
+    if (!metaClient.isMetadataTable()) { // lets validate data table that timestamps are generated in monotically increasing order. 
+      HoodieTableMetaClient reloadedMetaClient = HoodieTableMetaClient.reload(metaClient);
```

Comment:
The reloading of the timeline is costly, do we have better solution to avoid this.

Reply:
nope. unfortunately, we do not. We could avoid reloading, but that would defeat the purpose of guaranteeing latest commit time for any new commits when compared to other timestamps.

Reply:
may we we could do one optimization. 
For a strictly single writer use-case, we can avoid doing the timestamp validation and so we can avoid the latency hit. but when async table services or multi-writers are involved, we had to reload to validate.

Reply:
responded else where. 
again, do we really need those optimization? I dob't want to add such adhoc changes in 0.x branch just for optimization. This patch guards correctness and ensure data consistency. But any further optimization, since we are only targetting this for 0.x, I would like to be judicious.